cpp/src/arrow/util/value_parsing.h

```diff
@@ -281,6 +371,19 @@ struct StringToUnsignedIntConverterMixin {
     if (ARROW_PREDICT_FALSE(length == 0)) {
       return false;
     }
+    // If its starts with 0x then its hex
+    if (*s == '0' && *(s + 1) == 'x'){
+      length -= 2;
+      s += 2;
+      // lets make sure that the length of the string is not too big
```

Comment:
Nit: You can probably not need to decrease `length` and consider the offset of `s` as the difference to consider.

Reply:
I opted not to do this, since length is used both at StringToSignedIntConverterMixin and at ParseHex. Getting rid of length would then involve keeping track of the original pointer of `s` and the new pointer and comparing that to the original length. It seems more convoluted